### PR TITLE
Make cache-write failures non-fatal

### PR DESCRIPTION
**Pull request by charliermarsh**
## Summary

Closes https://github.com/astral-sh/ruff/issues/12284.

### CODE REVIEW

**Comment by charliermarsh** [comment]
Perhaps we should either (1) make these failure in debug or (2) gate this on a smaller subset of exceptions that we might "expect" (like access denied on Windows).

**Comment by charliermarsh** [comment]
One sec, making some changes.

File: crates/ruff/src/cache.rs [hunk 1]
```diff
@@ -180,12 +180,22 @@ impl Cache {
             .write_all(&serialized)
             .context("Failed to write serialized cache to temporary file.")?;
 
-        temp_file.persist(&self.path).with_context(|| {
-            format!(
-                "Failed to rename temporary cache file to {}",
-                self.path.display()
-            )
-        })?;
+        if let Err(err) = temp_file.persist(&self.path) {
```
Comment (charliermarsh) [inline crates/ruff/src/cache.rs:183]: The `open` method on this struct already uses a similar pattern of warning (but ignoring) some errors.

**Review by zanieb — APPROVED** [review]

**Comment by zanieb** [comment]
Should we be retrying like we do in uv?

**Comment by charliermarsh** [comment]
I don't think we should in this case because it's not a guard against anti-virus stuff like with binaries; it's a guard against the file being in-use, and it's not clear how long we'd expect that to be true or whether we should expect it to resolve.

**Comment by jaraco** [comment]
Thanks for the fast fix (and the fast release cadence so I don't have to inquire about when we might get the fix)!

**Pull request merged**